### PR TITLE
Mapper Actions improvements

### LINKED ISSUE #36: [FEATURE] Index Mappings Improvements
**Is your feature request related to a problem?**

1. **_CreateIndexMappingsAction_** relies on index having all paths defined in mappings, before applying specific alias mappings topic. We need capability to apply alias mappings partially, only to fields which are currently defined in index.

2. **_GetIndexMappingsAction_** is just a wrapper around GET _mapping API.

3. **_MapperFacade_** has topics hardcoded in constructor.

**What solution would you like?**
1. Add new boolean parameter **_partial_** to **_CreateIndexMappingsAction_** to allow partial alias mappings apply. Default values should be true.

2. Refactor **_GetIndexMappingsAction_** to return index mappings which consist only of aliases which are applied through Create/Update Mappings Actions

REQUEST:

Params:
| property | description |
| --- | --- |
| **_index_name_** | Name of index |

```
GET _plugins/_security_analytics/mappings?index_name=my_test_index
```
RESPONSE:
```
{
	"my_test_index":{
		"mappings":{
			"properties":{
				"source":{
					"properties":{
                                                  "source.ip": {
                                                          "type": "alias",
                                                          "path": "netflow.source_ipv4_address"
                                                  },
                                                  "source.port": {
                                                          "type": "alias",
                                                          "path": "netflow.source_transport_port"
                                                  }
					}
				}
			}
		}
	}
}
```

3. Refactor **_MapperFacade_** to load topics from json config file

### PR DESCRIPTION
**Pull request by petardz**
### Description
Extended capability of CreateMappingAction to allow partial mapping apply, in case index mappings are missing some of the alias mappings. This is controlled by request boolean property **_partial_**

Reworked GetMappingAction to return only alias mappings which are applied previously by Create/Update Mapping Actions.
 
### Issues Resolved
#36
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by getsaurabh02 — CHANGES_REQUESTED** [review]
Thanks @petardz LGTM, few comments to address.

File: src/main/java/org/opensearch/securityanalytics/action/GetIndexMappingsRequest.java [hunk 1]
```diff
@@ -50,6 +50,9 @@ public void writeTo(StreamOutput out) throws IOException {
     public static GetIndexMappingsRequest parse(XContentParser xcp) throws IOException {
         String indexName = null;
 
+        if (xcp.currentToken() == null) {
+            xcp.nextToken();
```
[the remark this answers is not in the document]
Comment (petardz) [reply src/main/java/org/opensearch/securityanalytics/action/GetIndexMappingsRequest.java:54]: For some reason I had to do this. currentToken was returning null.

File: src/main/java/org/opensearch/securityanalytics/mapper/MapperFacade.java [hunk 2]
```diff
@@ -4,20 +4,55 @@
...
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
+import java.util.stream.Collectors;
+
 public class MapperFacade {
+
+    private static final String MAPPER_CONFIG_FILE = "OSMapping/mapper_topics.json";
+
+    private static final Logger log = LogManager.getLogger(RestIndexDetectorAction.class);
```
[the remark this answers is not in the document]
Comment (petardz) [reply src/main/java/org/opensearch/securityanalytics/mapper/MapperFacade.java]: done

**Review by getsaurabh02 — APPROVED** [review]

**Comment by getsaurabh02** [comment]
Tests need fix @petardz

File: src/main/java/org/opensearch/securityanalytics/mapper/MappingsTraverser.java [hunk 3]
```diff
@@ -72,7 +78,23 @@ public MappingsTraverser(Map<String, Object> mappingsMap, Set<String> typesToSki
      */
     public MappingsTraverser(String mappings, Set<String> typesToSkip) throws IOException {
```
[the remark this answers is not in the document]
Comment (petardz) [reply src/main/java/org/opensearch/securityanalytics/mapper/MappingsTraverser.java:79]: added more javadocs

**Review by getsaurabh02 — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**